vllm/lora/models.py

```diff
@@ -119,32 +122,44 @@ def convert_mapping(
                                          dtype=torch.long)
     embeddings_indices = torch.stack([
         indices[2] * extra_vocab_size,
-        indices[2] * (vocab_size + extra_vocab_size)
+        indices[2] * (vocab_size + extra_vocab_size),
     ])
     embeddings_indices[embeddings_indices == -1] = max_loras - 1
     base_indices = indices[1]
     sampler_indices = prompt_mapping_tensor
     sampler_indices_padded = sampler_indices.clone()
     sampler_indices_padded[sampler_indices_padded == -1] = max_loras - 1
-    sampler_indices_padded = (
-        torch.arange(
-            0, len(sampler_indices_padded), device="cuda", dtype=torch.long) +
-        (sampler_indices_padded * len(sampler_indices_padded)))
+    sampler_indices_padded = torch.arange(
+        0, len(sampler_indices_padded), device="cuda", dtype=torch.long) + (
+            sampler_indices_padded * len(sampler_indices_padded))
     long_lora_indices = None
     long_lora_indices_len: Optional[int] = None
     if long_lora_context:
         long_lora_indices = indices[3]
         long_lora_indices_len = long_lora_indices.shape[-1]
     # Contain length of indices tensors. Used to index into each tensor.
     indices_len = [
-        base_indices.shape[-1], sampler_indices.shape[-1],
-        sampler_indices_padded.shape[-1], embeddings_indices.shape[-1]
+        base_indices.shape[-1],
+        sampler_indices.shape[-1],
+        sampler_indices_padded.shape[-1],
+        embeddings_indices.shape[-1],
     ]
     if long_lora_indices_len is not None:
         indices_len.append(long_lora_indices_len)
-
-    return (base_indices, sampler_indices, sampler_indices_padded,
-            embeddings_indices, long_lora_indices, indices_len)
+    else:
+        #If long_lora doesn't exist,append None
+        indices_len.append(None)
+    # Append a prefilling flag to help selecting the appropriate lora
+    # ops (sgmv or bgmv)
+    indices_len.append(int(mapping.is_prefilling))
+    return (
+        base_indices,
+        sampler_indices,
+        sampler_indices_padded,
+        embeddings_indices,
+        long_lora_indices,
+        indices_len,
```

Comment:
I am thinking it will be a good idea to change the return type of this to an object with some abstractions. Right now using `indices_len` is very hard because the developer needs to remember which index corresponds to what. We should abstract that away.

Reply:
I completely agree with this point. I had considered it before, but my concern was that changing the return value might necessitate modifying a lot of unit test code, so I left the function's return value unchanged.

Can we defer this modification to be completed in a future PR?

Reply:
Yeah we can do this in a followup